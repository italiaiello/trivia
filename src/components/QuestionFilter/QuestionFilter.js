import React, { useState } from 'react'
import Categories from './Filters/Categories'
import Difficulty from './Filters/Difficulty'
import Type from './Filters/Type'
import Quiz from '../Quiz/Quiz';

const QuestionFilter = () => {
    
    const [numQuestions, setNumQuestions] = useState(0)
    const [category, setCategory] = useState('Any')
    const [difficulty, setDifficulty] = useState('Any')
    const [type, setType] = useState('Any')

    const [route, setRoute] = useState('home')

    const onRouteChange = newRoute => {
        setRoute(newRoute)
    }

    const onNumQuestionsChange = e => setNumQuestions(e.target.value)

    return (
        <section>
            {
                route === 'home' ?
                <article>
                    <form className="questionFilter">
                        <input  type="text" 
                                placeholder="Number of questions (max 50)" 
                                onChange={onNumQuestionsChange} 
                        />
                        <Categories setCategory={setCategory} />
                        <Difficulty setDifficulty={setDifficulty} />
                        <Type setType={setType} />
                        <button onClick={() => onRouteChange('quiz')}>Submit</button>
                    </form>
                </article>
                :
                <Quiz   numQuestions={numQuestions} 
                        category={category} 
                        difficulty={difficulty} 
                        type={type}
                />
            }
        </section>
    )
}

export default QuestionFilter