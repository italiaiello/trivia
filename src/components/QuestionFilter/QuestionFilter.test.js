import React from 'react'
import { shallow } from 'enzyme'
import QuestionFilter from './QuestionFilter'

it('checks if it QuestionFilter.js renders properly', () => {

    expect(shallow(<QuestionFilter />)).toMatchSnapshot()
})