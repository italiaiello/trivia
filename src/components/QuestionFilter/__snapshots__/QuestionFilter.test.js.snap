// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if it QuestionFilter.js renders properly 1`] = `ShallowWrapper {}`;
